astroquery/mast/core.py

```diff
@@ -157,241 +155,101 @@ def __init__(self, username=None, password=None, session_token=None):
+        if token is None or reenter_token:
+            info_msg = "If you do not have an API token already, visit the following link to create one: "
+            log.info(info_msg + auth_link)
+            token = getpass("Enter MAST API Token: ")
 
-            if not value:
-                raise LoginError("Invalid session token")
+        # store token if desired
+        if store_token:
+            keyring.set_password("astroquery:mast.stsci.edu.token", "masttoken", token)
 
-            # add cookie to session
-            self._session.cookies.set(name, value)
+        self._session.headers["Accept"] = "application/json"
+        self._session.cookies["mast_token"] = token
+        info = self.session_info(silent=True)
 
+        if not info["anon"]:
+            log.info("MAST API token accepted, welcome %s" % info["attrib"].get("display_name"))
         else:
-            # raise datatype error
-            raise LoginError("Session token must be given as a dictionary or http.cookiejar.Cookie object")
+            warn_msg = ("MAST API token invalid!\n"
+                        "To make create a new API token visit to following link: " +
+                        auth_link)
+            warnings.warn(warn_msg, AuthenticationWarning)
 
-        # Print session info
-        # get user information
-        response = self._session.request("GET", self._SESSION_INFO_URL)
-
-        if response.status_code != 200:
-            warnings.warn("Status code: {}\nAuthentication failed!".format(response.status_code),
-                          AuthenticationWarning)
-            return False
-
-        exp = re.findall(r'<strong>Session Expiration \(barring inactivity\):</strong> (.*?)\n', response.text)
-        if len(exp) == 0:
-            warnings.warn("{}\nAuthentication failed!".format(response.text),
-                          AuthenticationWarning)
-            return False
-        else:
-            exp = exp[0]
+        return not info["anon"]
 
-        log.info("Authentication successful!\nSession Expiration: {}".format(exp))
-        return True
+    @deprecated(since="v0.3.10", message=("The get_token function is deprecated, "
+                                          "session token is now the token used for login."))
+    def get_token(self):
+        return None
 
-    def _shib_login(self, username, password):  # pragma: no cover
+    def session_info(self, silent=False):
         """
-        Given username and password, logs into the MAST shibboleth client.
+        Displays information about current MAST user, and returns user info dictionary.
 
         Parameters
         ----------
-        username : string
-            The user's username, will usually be the users email address.
-        password : string
-            Password associated with the given username.
-        """
-
-        # clear any previous shib cookies
-        self._session.cookies.clear_session_cookies()
+        silent : bool, optional
```

Comment:
We tend to have a `verbose` kwarg instead of `silent`, I think it would make sense to make it consistent.

Reply:
I agree. I'm trying to figure out the smoothest way to make this transition since `silent=False` == `verbose=True`. Does it make sense to have both arguments for a period of time where the use of "silent" triggers a deprecation warning?

Reply:
yes, and in fact there is a deprecation decorator in astropy to do this renaming: http://docs.astropy.org/en/latest/api/astropy.utils.decorators.deprecated_renamed_argument.html#astropy.utils.decorators.deprecated_renamed_argument

Reply:
Maybe I am missing an option, which would be great, but it looks to me like the two arguments need to **mean** the same thing to use that decorator. When in this case, to get the same behavior the user needs the opposite input.

Reply:
oh, yes, I suppose you're right. That case it can be deprecated for removal, and adding the verbose as a new one separately of the decorator. This very same decorator, with `new_name=None`.
And a short conditional to check for the case when both specified.

Reply:
Did it. The way decorating for removal appears to work is that the deprecated argument just doesn't get passed into the function anymore.  So that's why I didn't at the conditional check.

Reply:
I updated the decorator to put the info about the new argument to use in the 'since' argument. It's a bit hacky, but it means that it works with older versions of astropy utils now. And this won't be around forever.

Reply:
Still didn't work because "None" is not an acceptable new_name in the older version, so I pulled out the decorator and did the deprecation manually.  Not as tidy, but now it should work, and again the code won't be there forever.